lib/seahorn/BmcPass.cc

```diff
@@ -200,6 +205,18 @@ class BmcPass : public llvm::ModulePass {
         sem = std::make_unique<BvOpSem>(efac, *this,
                                         F.getParent()->getDataLayout(), MEM);
 
+      if (HornBv2 && UseCrabInferRng) {
+        // crab stuff
+        const llvm::Module &M = *(F.getParent());
+        // Get seadsa -- pointer analysis
+        auto &dsa_pass = getAnalysis<seadsa::ShadowMemPass>().getShadowMem();
+        auto &dsa = dsa_pass.getDsaAnalysis();
+
+        // XXX: use of legacy operational semantics
+        auto &tli = getAnalysis<TargetLibraryInfoWrapperPass>();
+        static_cast<Bv2OpSem &>(*sem).mkCrabAnalysis(M, dsa, tli);
```

Comment:
this seems wrong. why is crab exposed at this level. This should be inside opsem

Reply:
I see. So we can either use crab analysis in BvOpSem or BvOpSem2. Correct?

Reply:
Right, so that `BmcPass` can be used with any implementation of bmc and semantics, whether they use crab or not